src/deadline/houdini_submitter/python/deadline_cloud_for_houdini/submitter.py

```diff
@@ -274,44 +291,65 @@ def _get_job_template(rop: hou.Node) -> dict[str, Any]:
         # remove dependency info from name
         connected_node["name"] = connected_node["rop"]
         rop_steps = [connected_node]
-        ignore_input_nodes = "false"
+        ignore_input_nodes = False
     for node in rop_steps:
-        # init data
-        init_data_contents = []
-        init_data_contents.append("scene_file: '{{Param.HipFile}}'\n")
-        init_data_contents.append(f"render_node: '{node['rop']}'\n")
-        init_data_contents.append(f"version: {_get_houdini_version()}\n")
-        init_data_contents.append(f"ignore_input_nodes: {ignore_input_nodes}\n")
-        init_data_contents.append(f"wedgenum: '{node['wedgenum']}'\n")
-        init_data_contents.append(f"wedge_node: '{node['wedge_node']}'\n")
+        init_data = {
+            "scene_file": "{{Param.HipFile}}",
+            "render_node": node["rop"],
+            "version": _get_houdini_version(),
+            "ignore_input_nodes": ignore_input_nodes,
+            "wedgenum": node["wedgenum"],
+            "wedge_node": node["wedge_node"],
+        }
         init_data_attachment = {
             "name": "initData",
             "filename": "init-data.yaml",
             "type": "TEXT",
-            "data": "".join(init_data_contents),
+            # Convert to dict to YAML string using the prettier nested object format
+            "data": yaml.dump(init_data, default_flow_style=False),
         }
-        # environments
+
         environments = get_houdini_environments(init_data_attachment)
-        # task run data
-        task_data_contents = []
-        task_data_contents.append(f"render_node: {node['rop']}\n")
-        task_data_contents.append("frame: {{Task.Param.Frame}}\n")
-        task_data_contents.append(f"ignore_input_nodes: {ignore_input_nodes}\n")
-        # step
-        frame_range = "{start}-{stop}:{step}".format(**node)
+
+        task_data_dict = {"render_node": node["rop"], "ignore_input_nodes": ignore_input_nodes}
+
+        if node["render_strategy"] == RenderStrategy.SEQUENTIAL:
+            # Generate a single task that renders the whole frame range
+            parameter_space = None
+            task_data_dict["frame_range"] = {
+                "start": node["start"],
+                "end": node["end"],
+                "step": node["step"],
+            }
+        else:
+            # Generate one task per frame using step parameters
+            range_expression = "{start}-{end}:{step}".format(**node)
+            parameter_space = {
+                "taskParameterDefinitions": [
+                    {"name": "Frame", "range": range_expression, "type": "INT"}
+                ]
+            }
+            task_data_dict["frame_range"] = {
+                "start": "{{Task.Param.Frame}}",
+                "end": "{{Task.Param.Frame}}",
+                "step": 1,
+            }
+
+        task_data = yaml.dump(task_data_dict, default_flow_style=False)
```

Comment:
Does this end up changing what the exported templates look like? Based on the test it appears to be the same. Hmm... the readability hasn't been good there in the past. We should be thinking about turning this into a multiline yaml string, but that's future change and there's other things needed to make these generated templates better so just ignore this for now.

Also I haven't used pyyaml all too much, should we also be doing `safe_dump` here?

Reply:
This change does not affect the export templates except for reordering the fields. It produces the same format. I made this change to improve readability and to handle special characters (e.g. `"`) correctly.

Looking at the docs, I don't think we need `safe_dump`, but it produces the same result so I'll swap it out.

Reply:
Yea, I was on the fence about that based on the inputs we accept from the user. Thanks!